### PR TITLE
rename get_{ref, mut} to assume_init_{ref,mut} in Maybeuninit

### PR DESCRIPTION
**Pull request by Dylan-DPC-zz**
References #63568 

Rework with comments addressed from #66174 

Have replaced most of the occurrences I've found, hopefully didn't miss out anything 

r? @RalfJung 

(thanks @danielhenrymantilla for the initial work on this)

### CODE REVIEW

File: library/core/src/fmt/float.rs [hunk 1]
```diff
@@ -28,8 +28,8 @@ where
             *num,
             sign,
             precision,
-            buf.get_mut(),
-            parts.get_mut(),
+            buf.assume_init_mut(),
+            parts.assume_init_mut(),
```
Comment (RalfJung) [inline library/core/src/fmt/float.rs:32]: The FIXME above needs to also be adjusted.

Comment (RalfJung) [reply library/core/src/fmt/float.rs:32]: Also looks like the issue this is referencing got closed, and the new tracking issue does not even raise this question any more?

File: library/core/src/mem/maybe_uninit.rs [hunk 2]
```diff
@@ -167,7 +167,8 @@ use crate::mem::ManuallyDrop;
 ///
 /// // For each item in the array, drop if we allocated it.
 /// for elem in &mut data[0..data_len] {
-///     unsafe { ptr::drop_in_place(elem.as_mut_ptr()); }
+///     unsafe { ptr::drop_in_place(elem.
+/// ptr()); }
```
Comment (RalfJung) [inline library/core/src/mem/maybe_uninit.rs:171]: Spurious newline?

File: library/core/src/mem/maybe_uninit.rs [hunk 3]
```diff
@@ -716,7 +717,7 @@ impl<T> MaybeUninit<T> {
     /// fn read_chunk (reader: &'_ mut dyn io::Read) -> io::Result<[u8; 64]>
     /// {
     ///     let mut buffer = MaybeUninit::<[u8; 64]>::uninit();
-    ///     reader.read_exact(unsafe { buffer.get_mut() })?;
+    ///     reader.read_exact(unsafe { buffer.assume_init_mut() })?;
     ///                             // ^^^^^^^^^^^^^^^^
```
Comment (RalfJung) [inline library/core/src/mem/maybe_uninit.rs:721]:
```suggestion
    ///     reader.read_exact(unsafe { buffer.assume_init_mut() })?;
    ///                             // ^^^^^^^^^^^^^^^^^^^^^^^^
```
and likewise below

File: library/std/src/io/util.rs [hunk 4]
```diff
@@ -55,18 +55,18 @@ where
     // to uninitialized data, but within libstd we can rely on more guarantees
     // than if this code were in an external lib.
     unsafe {
-        reader.initializer().initialize(buf.get_mut());
+        reader.initializer().initialize(buf.assume_init_mut());
```
Comment (RalfJung) [inline library/std/src/io/util.rs:58]: Same FIXME as above.

File: library/core/src/fmt/float.rs (hunk 1 above)
Comment (Dylan-DPC) [reply library/core/src/fmt/float.rs:32]: Created https://github.com/rust-lang/rust/issues/76092

File: library/core/src/fmt/float.rs [hunk 5]
```diff
@@ -47,19 +47,19 @@ fn float_to_decimal_common_shortest<T>(
...
     T: flt2dec::DecodableFloat,
 {
-    // SAFETY: Possible undefined behavior, see FIXME(#53491)
+    // SAFETY: Possible undefined behavior, see FIXME(#76092)
     unsafe {
         // enough for f32 and f64
         let mut buf = MaybeUninit::<[u8; flt2dec::MAX_SIG_DIGITS]>::uninit();
         let mut parts = MaybeUninit::<[flt2dec::Part<'_>; 4]>::uninit();
-        // FIXME(#53491)
+        // FIXME(#76092)
```
Comment (RalfJung) [inline library/core/src/fmt/float.rs:55]: Could you grep the code for `53491` to make sure there are no other instances of this left?

File: library/core/src/mem/maybe_uninit.rs [hunk 6]
```diff
@@ -624,14 +624,14 @@ impl<T> MaybeUninit<T> {
     /// let b = MaybeUninit::<Cell<bool>>::uninit();
     /// // Initialize the `MaybeUninit` using `Cell::set`:
     /// unsafe {
-    ///     b.get_ref().set(true);
+    ///     b.assume_init_ref().set(true);
     ///  // ^^^^^^^^^^^
```
Comment (RalfJung) [inline library/core/src/mem/maybe_uninit.rs:628]: The `^` here do not match the code above any more-

**Comment by RalfJung** [comment]
Thanks a lot for picking this up @Dylan-DPC :) 
r=me with the two remaining minor nits resolved.

**Pull request merged**